src/functionaltests/resources/features/linking.feature

```diff
@@ -0,0 +1,97 @@
+    And CHANGE links to event "SCC1"
+
+  Scenario: Normal scenario with one change
+    Given a SCS event with id "SCS1" was sent on "refs/for/master"
+
+    When user "1" creates a new change on "refs/for/master"
+    Then a "SCC" event with id "SCC1" is sent
+    And BASE links to event "SCS1"
+
+    When user "1" submits the change to "refs/for/master"
+    Then a "SCS" event with id "SCS2" is sent
+    And CHANGE links to event "SCC1"
+    And PREVIOUS_VERSION links to event "SCS1"
+
+  Scenario: Normal scenario with two changes
+    Given a SCS event with id "SCS1" was sent on "refs/for/master"
+
+    When user "1" creates a new change on "refs/for/master"
+    Then a "SCC" event with id "SCC1" is sent
+    And BASE links to event "SCS1"
+
+    When user "1" uploads a new patchset to "refs/for/master"
+    Then a "SCC" event with id "SCC2" is sent
+    And BASE links to event "SCS1"
+    And PREVIOUS_VERSION links to event "SCC1"
+
+    When user "1" submits the change to "refs/for/master"
+    Then a "SCS" event with id "SCS2" is sent
+    And CHANGE links to event "SCC2"
+    And PREVIOUS_VERSION links to event "SCS1"
+
+  @Test1
+  Scenario: Multiple patch sets created simultaneously
+    Given a SCS event with id "SCS1" was sent on "refs/for/master"
+
+    When user "1" creates a new change on "refs/for/master"
+    Then a "SCC" event with id "SCC1" is sent
+    And BASE links to event "SCS1"
+
+    When user "2" creates a new change on "refs/for/master"
+    Then a "SCC" event with id "SCC2" is sent
+    And BASE links to event "SCS1"
+
+    When user "3" creates a new change on "refs/for/master"
+    Then a "SCC" event with id "SCC3" is sent
+    And BASE links to event "SCS1"
+
+    When user "2" submits the change to "refs/for/master"
+    Then a "SCS" event with id "SCS2" is sent
+    And CHANGE links to event "SCC2"
+    And PREVIOUS_VERSION links to event "SCS1"
+
+    When user "1" uploads a new patchset to "refs/for/master"
+    Then a "SCC" event with id "SCC4" is sent
+    And BASE links to event "SCS1"
+    And PREVIOUS_VERSION links to event "SCC1"
+
+    When user "3" rebases the change for "refs/for/master"
+    And user "3" submits the change to "refs/for/master"
+    Then a "SCS" event with id "SCS3" is sent
+    And CHANGE links to event "SCC3"
+    And PREVIOUS_VERSION links to event "SCS2"
+
+    When user "1" rebases the change for "refs/for/master"
+    And user "1" submits the change to "refs/for/master"
+    Then a "SCS" event with id "SCS4" is sent
+    And CHANGE links to event "SCC4"
+    And PREVIOUS_VERSION links to event "SCS3"
+
+  Scenario: Normal scenario submitting refs/for/other-branch than refs/for/master
+    Given a SCS event with id "SCS1" was sent on "refs/for/master"
+    And a SCS event with id "SCS2" was sent on "refs/for/other-branch"
+
+    When user "1" creates a new change on "refs/for/other-branch"
+    Then a "SCC" event with id "SCC1" is sent
+    And BASE links to event "SCS2"
+
```

Comment:
It would also be good if we can explain the scenarios keeping multiple branches in mind, like

a new patchset is pushed to master, and then it was submitted, what happens with this link model and so on.

Reply:
Fixed in 5225987